### PR TITLE
Add test for `selectable_region.0.dart`

### PR DESCRIPTION
**Pull request by ValentinVignal**
Contributes to https://github.com/flutter/flutter/issues/130459

It adds a test for
- `examples/api/test/material/selectable_region/selectable_region.0_test.dart`

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

File: examples/api/lib/material/selectable_region/selectable_region.0.dart [hunk 1]
```diff
@@ -64,23 +64,24 @@ class _SelectableAdapter extends SingleChildRenderObjectWidget {
...
     renderObject
       ..selectionColor = DefaultSelectionStyle.of(context).selectionColor!
       ..registrar = registrar;
   }
 }
 
-class _RenderSelectableAdapter extends RenderProxyBox with Selectable, SelectionRegistrant {
-  _RenderSelectableAdapter(
+@visibleForTesting
+class RenderSelectableAdapter extends RenderProxyBox with Selectable, SelectionRegistrant {
```
Comment (ValentinVignal) [inline examples/api/lib/material/selectable_region/selectable_region.0.dart:83]: Is there a better way to test the custom selection?

File: examples/api/test/material/selectable_region/selectable_region.0_test.dart [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+  @override
+  bool matches(Element candidate) {
+    return candidate.renderObject.runtimeType == renderObjectType;
+  }
+}
+
+extension on CommonFinders {
+    Finder byRenderObjectType(Type renderObjectType, { bool skipOffstage = true }) => _RenderObjectTypeWidgetFinder(renderObjectType, skipOffstage: skipOffstage);
+}
```
Comment (ValentinVignal) [inline examples/api/test/material/selectable_region/selectable_region.0_test.dart:84]: Is there a better of doing this? If not, do we want it to be added with the other default finders in `packages/flutter_test/lib/src/finders.dart` ?

File: examples/api/lib/material/selectable_region/selectable_region.0.dart [hunk 3]
```diff
@@ -322,6 +322,8 @@ class _RenderSelectableAdapter extends RenderProxyBox with Selectable, Selection
   @override
   void dispose() {
     _geometry.dispose();
+    _startHandle = null;
+    _endHandle = null;
```
Comment (bleroux) [inline examples/api/lib/material/selectable_region/selectable_region.0.dart:326]: Great catch!!!
Maybe that's one of the reason adding test is useful :rofl:

File: examples/api/test/material/selectable_region/selectable_region.0_test.dart [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    LogicalKeyboardKey key,
+  ) async {
+    final LogicalKeyboardKey modifier = switch (defaultTargetPlatform) {
+      TargetPlatform.iOS || TargetPlatform.macOS => LogicalKeyboardKey.meta,
+      _ => LogicalKeyboardKey.control,
+    };
+    await tester.sendKeyDownEvent(modifier);
+    await tester.sendKeyDownEvent(key);
+    await tester.sendKeyUpEvent(key);
+    await tester.pump();
```
Comment (bleroux) [inline examples/api/test/material/selectable_region/selectable_region.0_test.dart:23]: I tried to put the pump at the end and it works.
Not sure why `sendKeyCombination` from packages/flutter/test/widgets/keyboard_utils.dart does not pump at the end.
No strong opinion on this, feel free to keep it as-is or move the pump.

File: examples/api/test/material/selectable_region/selectable_region.0_test.dart [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    await tester.tap(find.byIcon(Icons.key)); // Focus the application.
+    await tester.pump();
+
+    // Keyboard select all.
+    await sendKeyCombination(tester,LogicalKeyboardKey.keyA);
+    // Keyboard copy.
+    await sendKeyCombination(tester,LogicalKeyboardKey.keyC);
+
+    expect(clipboard, 'SelectableRegion SampleSelect this iconCustom Text');
+    }, variant: TargetPlatformVariant.all());
```
Comment (bleroux) [inline examples/api/test/material/selectable_region/selectable_region.0_test.dart:52]:
```suggestion
  }, variant: TargetPlatformVariant.all());
```

**Review by bleroux — APPROVED** [review]
LGTM! Nice test addition :+1:

File: examples/api/test/material/selectable_region/selectable_region.0_test.dart (hunk 5 above)
Comment (ValentinVignal) [reply examples/api/test/material/selectable_region/selectable_region.0_test.dart]: Oops, I removed it in [🎨 Move pump and remove extra white spaces](https://github.com/flutter/flutter/pull/157258/commits/e298e15b75be554e10efdf682029bf6f2331a16e)

**Pull request merged**